### ISSUE TITLE
[TT-Train] SoftmaxBackward ManyShortRows/fp32 unit test fails

### ISSUE THREAD

**Issue by vtsilytskyiTT** [tenstorrent/tt-metal#41944]
Reproduction steps:
1. Open `tt-train/tests/ops/softmax_backward_op_test.cpp`
2. Set name `.name = "many_short_rows_non_streaming_5_tiles",` (it impacts the seed) in `NIGHTLY_SoftmaxBackward_ManyShortRows`
3. Build tt-train `./build_metal.sh --build-tt-train --build-tests --debug`
4. Run this test `./build/tt-train/tests/ttml_tests --gtest_filter='SoftmaxBackward/SoftmaxBackwardOpTypedTest.NIGHTLY_SoftmaxBackward_ManyShortRows2/fp32'`

Expected: test pass.
Actual: test fails with **max_abs_diff=1.06631**:

```
Note: Google Test filter = SoftmaxBackward/SoftmaxBackwardOpTypedTest.NIGHTLY_SoftmaxBackward_ManyShortRows2/fp32
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from SoftmaxBackward/SoftmaxBackwardOpTypedTest
2026-04-13 10:25:24.944 | warning  |          Always | Unknown motherboard 'H12DSG-O-CPU' for chip_id=0 (bus_id=0x25) — defaulting tray_id to 0. Add this motherboard and its bus IDs to mobo_to_bus_ids in physical_system_discovery.cpp. (physical_system_discovery.cpp:62)
[ RUN      ] SoftmaxBackward/SoftmaxBackwardOpTypedTest.NIGHTLY_SoftmaxBackward_ManyShortRows2/fp32
[SoftmaxBackward test seeds] case="many_short_rows_non_streaming_5_tiles" dtype=fp32 kSuiteSeed=42 logits_seed=3241431015 grad_seed=1788919540
2026-04-13 10:25:39.242 | debug    |              Op | Launching Device Operation: "SoftmaxBackwardDeviceOperation" (device_operation.hpp:150)
2026-04-13 10:25:39.242 | debug    |              Op | Program Hash: 18446744073360812377 (device_operation.hpp:152)
2026-04-13 10:25:39.242 | debug    |              Op | Program Cache Hit: false (device_operation.hpp:153)
2026-04-13 10:25:39.242 | debug    |              Op | Attributes: (device_operation.hpp:155)
2026-04-13 10:25:39.242 | debug    |              Op |  dim = 3 (device_operation.hpp:157)
2026-04-13 10:25:39.242 | debug    |              Op |  sub_core_grids = std::nullopt (device_operation.hpp:157)
2026-04-13 10:25:39.242 | debug    |              Op | Tensors Args: (device_operation.hpp:160)
2026-04-13 10:25:39.243 | debug    |              Op |  0: Tensor(storage=DeviceStorage(),tensor_spec=TensorSpec(logical_shape=Shape([1, 30, 6400, 160]),tensor_layout=TensorLayout(dtype=DataType::FLOAT32,page_config=PageConfig(config=TilePageConfig(tile=Tile(tile_shape={32, 32},face_shape={16, 16},num_faces=4))),memory_config=MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt,nd_shard_spec=std::nullopt,created_with_nd_shard_spec=0),alignment=Alignment([32, 32])))) (device_operation.hpp:164)
2026-04-13 10:25:39.243 | debug    |              Op |  1: Tensor(storage=DeviceStorage(),tensor_spec=TensorSpec(logical_shape=Shape([1, 30, 6400, 160]),tensor_layout=TensorLayout(dtype=DataType::FLOAT32,page_config=PageConfig(config=TilePageConfig(tile=Tile(tile_shape={32, 32},face_shape={16, 16},num_faces=4))),memory_config=MemoryConfig(memory_layout=TensorMemoryLayout::INTERLEAVED,buffer_type=BufferType::DRAM,shard_spec=std::nullopt,nd_shard_spec=std::nullopt,created_with_nd_shard_spec=0),alignment=Alignment([32, 32])))) (device_operation.hpp:164)
2026-04-13 10:25:39.243 | debug    |              Op |  (device_operation.hpp:169)
2026-04-13 10:25:39.243 | debug    |              Op | SoftmaxBackward: Using NON-STREAMING mode | Shape: 6000x5 tiles | 2x buffering | Estimated L1: 88 KB (softmax_backward_program_factory.cpp:219)
/proj_sw/user_dev/vtsilytskyi/tt-metal/tt-train/tests/ops/softmax_backward_op_test.cpp:177: Failure
Value of: close
  Actual: false
Expected: true
case=many_short_rows_non_streaming_5_tiles, max_abs_diff=1.06631, atol=0.025, rtol=0.025, worst_elem(n,c,h,w)=(0,0,5366,106) flat_idx=858666 result=0.61964 expected=-0.446673 abs_diff=1.06631
[  FAILED  ] SoftmaxBackward/SoftmaxBackwardOpTypedTest.NIGHTLY_SoftmaxBackward_ManyShortRows2/fp32, where GetParam() = 16-byte object <B2-CE 6B-61 C5-55 00-00 01-00 00-00 01-00 00-00> (22037 ms)
[----------] 1 test from SoftmaxBackward/SoftmaxBackwardOpTypedTest (22037 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (28661 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] SoftmaxBackward/SoftmaxBackwardOpTypedTest.NIGHTLY_SoftmaxBackward_ManyShortRows2/fp32, where GetParam() = 16-byte object <B2-CE 6B-61 C5-55 00-00 01-00 00-00 01-00 00-00>

 1 FAILED TEST
```

**Comment by bbradelTT** [comment]
This is tt-train operation, not a ttnn one.
@dgomezTT please triage/re-assign.

**Comment by dgomezTT** [comment]
Resolved by merged PR #41947 ("[ttml] softmax_backward FP32 support"), which adds the missing FP32 path that the failing ManyShortRows/fp32 unit test exercises.
